backend/orchestrators/Orchestrator.py

```diff
@@ -286,7 +287,7 @@ def prepare_body_headers_with_data(self, request):
 
         headers = {
             'Content-Type': 'application/json',
-            'api-key': self.AZURE_OPENAI_KEY,
+            'api-key': key,
```

Comment:
insert the appropriate key into the header according to what was selected above.